grails-app/services/org/pih/warehouse/inventory/CycleCountService.groovy

```diff
@@ -528,7 +528,9 @@ class CycleCountService {
         cycleCountItem.properties = command.properties
         cycleCountItem.countIndex = command.recount ? 1 : 0
         cycleCountItem.status = command.recount ? CycleCountItemStatus.INVESTIGATING : CycleCountItemStatus.COUNTING
-        cycleCountItem.dateCounted = new Date()
+        if (cycleCountItem.dateCounted = null) {
+            cycleCountItem.dateCounted = new Date()
```

Comment:
I don't really understand this part. This code now suggests that we always set the date for current date, no matter what we provide in the command.

Reply:
We already have `cycleCountItem.properties = command.properties` so all this change does is only set the date counted as today if the client isn't specifying it themselves and the item doesn't already have a date counted set

Reply:
What if the `dateCounted` is provided as `null` in the command? Wouldn't the `cycleCountItem.properties = command.properties` override the date to be null? @SebastianLib have you checked such case?
I want to make sure this won't actually cause the problem mentioned by you @EWaterman here: https://github.com/openboxes/openboxes/pull/5139#discussion_r2001574078

By saying so, this is why I was against setting the properties via `.properties =`. If we set it expliciltly, we would immediately see which properties we mutate and which not.

Reply:
good point. If the frontend is explicitly unsetting the dateCounted field (by providing dateCounted: null), we should respect that. Actually, since this is an update API, maybe we shouldn't be trying to provide a new Date() fallback at all. It's safe to assume that the default we provide when creating a new row is good enough.

In that case, we can simply remove this whole if block. Thoughts?

(And yeah I'm not sure how I feel about the `.properties =` solution. On the one hand, it's a good reminder that the API should be a simple update on count item fields with no complex logic, but on the other hand, it's less explicit about what it's doing and so can be somewhat obtuse.)